### DISCUSSION TITLE
branch for customize nightscout on google cloud nightscout?

### DISCUSSION THREAD

**Discussion by tynbendad** [NightscoutFoundation/xDrip#2987] [General]
i tried using stock nightscout/cgm-remote-monitor master branch for the customize nightscout option but it just gives "502 Bad Gateway" - site worked fine with the stock google cloud nightscout install.  is there a specific branch i need to use? i'd rather use something based on 14.2.6 than dev branch..

**Comment by Navid200** [comment]
Unfortunately, if you want to customize Nightscout, you need to use a branch that is based on 15, not 14.2.6.
This limitation was not put in place intentionally.  But, after I discovered it, I have not heard any requests.
You don't have to use the latest development branch being very unstable if that's your concern.
You can use any branch based on any version of 15.  So, you can start from the earliest branch and build from it.

If you can tell me exactly what your concern is and if I understand that there is a legitimate reason, I am open to spending time to address it.

**Reply by tynbendad** [reply]
Thanks, do you know the earliest 15 branch that is stable? My concern is I will be using this site, not for development. I have some minor changes already made on 14.2.6 and I don't mind updating them to work with 15 but I need a stable daily driver site.

**Reply by Navid200** [reply]
I'm sorry I don't know what you mean by the earliest that is "stable".

The development branch has not changed since January.  
Why don't you start from the current development branch, and copy the files you have changed to it.  If it works, you are in business, no?
If it doesn't then, we can talk.

Or, am I missing something?

**Reply by tynbendad** [reply]
I need something that I know works well in order to use the site as our main site. I don't know anything about the state of the dev branch...

**Reply by tynbendad** [reply]
also i'm not sure which dev branch you are saying hasn't changed since january? - where i'm looking has lots of commits since january: https://github.com/nightscout/cgm-remote-monitor/commits/dev

**Reply by Navid200** [reply]
My mistake!

This is what I used to create my branch in order to customize:
https://github.com/nightscout/cgm-remote-monitor/commits/9258cdc0e75dcd4ebcec6afdc1e4392181f31d9b

I have been using it since then (May, 2023) and it has been fine for my use.
But, that doesn't mean it will be fine for you.  Only you can tell.

**Reply by tynbendad** [reply]
i created a branch from dev branch at the commit in your link and it still gave 502 error (my branch is https://github.com/tynbendad/cgm-remote-monitor/tree/dev-2may23-unchanged)... then i created a branch from today's dev branch and its working. i guess i can't really use it until i test it a while but its something.
